Makefile

```
@@ -1,6 +1,20 @@
 GOOS ?= $(shell go env GOOS)
 SOURCES := $(shell find . -type f  -name '*.go')
-LDFLAGS := ""
+
+# Git information
+GIT_VERSION ?= $(shell git describe --always --dirty)
```

By default (without --all or --tags) `git describe` only shows annotated tags.  But the maintainer usually creates new releases from GitHub that not support annotate tags. So how about this? 
```suggestion
GIT_VERSION ?= $(shell git describe --tags --dirty)
```

---

E.g:
```
# ./karmadactl version
karmadactl version: version.Info{GitVersion:"v0.5.0-12-g8f338a0", GitCommit:"8f338a0aa793ce01b6fa152f15298fcb26cd33b6", GitTreeState:"clean", BuildDate:"2021-04-26T16:36:13Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```

---

> By default (without --all or --tags) `git describe` only shows annotated tags. But the maintainer usually creates new releases from GitHub that not support annotate tags. So how about this?

Okay, That's a good idea.